processor/src/test/java/org/mapstruct/ap/test/bugs/_1541/Issue1541Test.java

```diff
@@ -0,0 +1,112 @@
+/**
+ *  Copyright 2012-2017 Gunnar Morling (http://www.gunnarmorling.de/)
+ *  and/or other contributors as indicated by the @authors tag. See the
+ *  copyright.txt file in the distribution for a full listing of all
+ *  contributors.
+ *
+ *  Licensed under the Apache License, Version 2.0 (the "License");
+ *  you may not use this file except in compliance with the License.
+ *  You may obtain a copy of the License at
+ *
+ *      http://www.apache.org/licenses/LICENSE-2.0
+ *
+ *  Unless required by applicable law or agreed to in writing, software
+ *  distributed under the License is distributed on an "AS IS" BASIS,
+ *  WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+ *  See the License for the specific language governing permissions and
+ *  limitations under the License.
+ */
+package org.mapstruct.ap.test.bugs._1541;
+
+import org.junit.Test;
+import org.junit.runner.RunWith;
+import org.mapstruct.ap.testutil.IssueKey;
+import org.mapstruct.ap.testutil.WithClasses;
+import org.mapstruct.ap.testutil.runner.AnnotationProcessorTestRunner;
+
+import static org.assertj.core.api.Assertions.assertThat;
+
+/**
+ *
+ * @author Christian Bandowski
+ */
+@WithClasses({
+    Issue1541Mapper.class,
+    Source.class,
+    Target.class
+})
+@RunWith(AnnotationProcessorTestRunner.class)
+@IssueKey("1541")
+public class Issue1541Test {
+
+    @Test
+    public void testMappingWithVarArgs() {
+        Target target = Issue1541Mapper.INSTANCE.mapWithVarArgs( "code", "1", "2" );
+
+        assertThat( target ).isNotNull();
+        assertThat( target.getCode() ).isEqualTo( "code" );
+        assertThat( target.getParameters() ).contains( "1", "2" );
+        assertThat( target.isAfterMappingWithArrayCalled() ).isFalse();
```

Comment:
Why are these false? By looking into the `@AfterMapping` I would assume that they are called. OK I see the `@BeanMapping` and `qualifiedByName` now. However, it seems a bit strange and not obvious when looking at the code.

Reply:
Yes that's the reason. The `@AfterMapping`s are only used when they are `qualifiedByName` (at first I also thought that they will be used automatically in case no `qualifiedByName` value is given.. but looks like an `@AfterMapping` is not automatically used when it is `@Named`)

Reply:
Yes when a qualifier is used they are explicitly ignored